akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileUploadDirectives.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.http.scaladsl.server.directives
+
+import java.io.File
+
+import akka.stream.scaladsl._
+import akka.http.scaladsl.server.Directive1
+import akka.http.scaladsl.model.{ ContentType, Multipart }
+import akka.util.ByteString
+import scala.util.Try
+
+trait FileUploadDirectives {
+
+  import BasicDirectives._
+  import FutureDirectives._
+  import MarshallingDirectives._
+
+  /**
+   * Collects the bytes of the first file part from the request into the given destination file
+   *
+   * @param destination Will be overwritten if not empty
+   */
+  def uploadedFile(destination: File): Directive1[Try[UploadedFile]] =
```

Comment:
I'm starting to think it is a bit weird to not talk about field name here, if we look at the Play API for the same thing for example that is kind of central: https://www.playframework.com/documentation/2.0.2/JavaFileUpload

Maybe it should be `uploadedFile(fieldName: String, destination: File)` rather?